### PR TITLE
Reorganize ops/utils and improve naming for BasicSensor

### PR DESCRIPTION
**Pull request by diptodip**
This PR brings the reorganization of ops and utils to make the separation more clear from PR #103 and also fixes the naming scheme for `BasicShotNoiseSensor -> BasicSensor` as discussed in Issue #20, along with a few other small bug fixes. Variables (i.e. attributes of elements that are not trainable parameters) are now initialized without passing a `None` as the first argument, which allows for example initializing a non-trainable phase mask with one of the phase mask initializers in `chromatix.utils.initializers`.

### CODE REVIEW

File: src/chromatix/elements/propagation.py [hunk 1]
```diff
@@ -46,9 +45,14 @@ class Propagate(nn.Module):
 
     then this element has a trainable refractive index, initialized to 1.33.
 
-    This element can cache the propagation kernel using the option
-    ``cache_propagator``. If you would like to have a trainable propagation
-    kernel, see ``KernelPropagate`` which accepts a trainable ``propagator``.
+    !! By default this element caches the propagation kernel using the option
```
Comment (GJBoth) [inline src/chromatix/elements/propagation.py:48]: Nice, this was necessary!

File: src/chromatix/elements/sources.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ class PlaneWave(nn.Module):
     spectral_density: Union[float, Array]
     power: Union[float, Callable[[PRNGKey], float]] = 1.0
     amplitude: Union[float, Array, Callable[[PRNGKey], Array]] = 1.0
-    kykx: Array = jnp.zeros((2,))
+    kykx: Array = (0.0, 0.0)
```
Comment (GJBoth) [inline src/chromatix/elements/sources.py:170]: Now the type is not an array anymore - why did you change to a tuple?

File: src/chromatix/functional/phase_masks.py [hunk 3]
```diff
@@ -39,15 +39,15 @@ def wrap_phase(phase: Array, limits: Tuple[float, float] = (-jnp.pi, jnp.pi)) ->
...
-        + 2 * jnp.pi * (1 + (phase_min - phase[min_indices]) // (2 * jnp.pi))
+    phase = jnp.where(
+        phase < phase_min,
+        phase + 2 * jnp.pi * (1 + (phase_min - phase) // (2 * jnp.pi)),
+        phase,
     )
-    phase = phase.at[max_indices].set(
-        phase[max_indices]
-        - 2 * jnp.pi * (1 + (phase[max_indices] - phase_max) // (2 * jnp.pi))
+    phase = jnp.where(
```
Comment (GJBoth) [inline src/chromatix/functional/phase_masks.py:47]: so much cleaner - nice!

File: src/chromatix/functional/propagation.py [hunk 4]
```diff
@@ -63,7 +64,7 @@ def transfer_propagate(
     n: float,
     N_pad: int,
     cval: float = 0,
-    kykx: Array = jnp.zeros((2,)),
+    kykx: Array = (0.0, 0.0),
```
Comment (GJBoth) [inline src/chromatix/functional/propagation.py:67]: Same as before, we need to change the type, and not sure we can parallelize over this as no longer an Array?

**Review by GJBoth — COMMENTED** [review]
Looks quite good to me! Only question I have is why change kykx from an Array to a tuple? That implies we can't parallelize over that anymore?

**Comment by GJBoth** [comment]
Good point! In that case, we just need to update the type annotation to `Array | tuple[float, float]` :-)

**Comment by diptodip** [comment]
> I feel slightly in favour of updating the type annotation, just so we're consistent...

Done!

**Review by GJBoth — APPROVED** [review]
Now everything looks good to me, thanks!
